src/multiplex/client.rs

```diff
@@ -46,7 +46,7 @@ where
     T: TryStream + Sink<Request> + TagStore<Request, T::Ok>,
 {
     /// Store the provided pending request.
-    fn sent(self: Pin<&mut Self>, pending: Pending<T::Tag, T::Ok>);
+    fn sent(self: Pin<&mut Self>, pending: Pending<T::Tag, T::Ok>, transport: Pin<&mut T>);
```

Comment:
Out of curiousity, why not passing the `PendingStore` to `MultiplexTransport` and make it `impl PendingStore`, as it is done for `TagStore` ?

Reply:
@jonhoo and I talked about it on slack, maybe he can chime in on why he wanted to keep T not overloaded with too many bounds.

Reply:
We could do that too — that'd be a separate change from this one. My thinking here was that I don't want to _require_ that the user have a _single_ type that implements all these traits.